torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -242,9 +242,48 @@ Value* TracingState::getOutput(const IValue& iv, size_t i) {
         fmap(tuple, [&](const IValue& ival) { return getOutput(ival, i); }));
     graph->insertNode(tuple_node);
     return tuple_node->output();
+  } else if (iv.isGenericDict()) {
+    if (tracing_mode_strict) {
+      throw std::runtime_error(
+          "Encountering a dict at the output of the tracer" +
+          std::string(STRICT_TRACER_MSG));
+    }
+    auto dict = iv.toGenericDict();
+    TypePtr key_type = dict.keyType();
+    TypePtr value_type = dict.valueType();
+
+    if (!(key_type->cast<StringType>() || key_type->cast<TensorType>()) ||
+        !value_type->cast<TensorType>()) {
+      std::ostringstream os;
+      os << "output " << i << " (" << dict << ") of traced region "
+         << "cannot be understood by the tracer, only dict[str, Tensor] "
+         << "or dict[Tensor, Tensor] can be a dictionary output of a traced function";
+      throw std::runtime_error(os.str());
+    }
+    const auto order = iterationOrder(dict);
+    std::vector<Value*> keys;
+    std::vector<Value*> values;
+    for (const auto& pair : order) {
+      keys.emplace_back(getValue(pair.first));
+      values.emplace_back(getOutput(pair.second, i));
+    }
+    if (dict.keyType()->cast<TensorType>()) {
+      // First get a permuation that are sorted by the keys Value's unique id
+      std::vector<int64_t> perm(keys.size());
+      std::iota(perm.begin(), perm.end(), 0);
+      std::sort(perm.begin(), perm.end(), [&](int64_t i, int64_t j) {
+        return keys[i]->unique() < keys[j]->unique();
+      });
```

Comment:
Can you comment on the stability of these unique IDs? Doesn't Value::unique() potentially depend on the order of compilation of values?